### PR TITLE
Dont send exclude pks through js, instead do the excluding in Python.

### PR DESCRIPTION
**Pull request by joeyjurjens**

### CODE REVIEW

File: wagtailautocomplete/views.py [hunk 1]
```diff
@@ -63,20 +64,34 @@ def search(request):
...
+    # Get exclusions based on the existing instance.
+    instance = request.GET.get('instance', '')
+    db_field = request.GET.get('db_field', '')
+    if instance and db_field:
+        model_str, field_str = db_field.rsplit(".", 1)
+
+        try:
+            model = apps.get_model(model_str)
+        except Exception:
+            return HttpResponseBadRequest()
```
Comment (specialunderwear) [inline wagtailautocomplete/views.py]: Please mention which thing is bad here.

File: wagtailautocomplete/views.py [hunk 2]
```diff
@@ -63,20 +64,34 @@ def search(request):
...
+
+        if isinstance(model._meta.get_field(field_str), ManyToManyField):
+            model_instance = model.objects.get(pk=instance)
+            model_field = getattr(model_instance, field_str)
+            exclusions = model_field.all().values_list("pk", flat=True).distinct()
```
Comment (specialunderwear) [inline wagtailautocomplete/views.py]: no need for distinct, it's just for the exclusion

File: wagtailautocomplete/views.py [hunk 3]
```diff
@@ -63,20 +64,34 @@ def search(request):
...
+            queryset = queryset.exclude(pk__in=exclusions)
+        else:
+            queryset = queryset.exclude(pk=instance)
```
Comment (specialunderwear) [inline wagtailautocomplete/views.py]: I don't think this is the correct query? shouldn't this do something with the db_field?

**Review by specialunderwear — CHANGES_REQUESTED** [review]

File: wagtailautocomplete/views.py (hunk 2 above)
Comment (joeyjurjens) [reply wagtailautocomplete/views.py]: Removed the distinct.

File: wagtailautocomplete/views.py (hunk 1 above)
Comment (joeyjurjens) [reply wagtailautocomplete/views.py]: Done.

File: wagtailautocomplete/views.py (hunk 3 above)
Comment (joeyjurjens) [reply wagtailautocomplete/views.py]: Yeah you were right, I've fixed the logic here.

**Pull request merged**